### PR TITLE
[unidown]: Lots of fixes from the sprint

### PR DESCRIPTION
**Pull request by ehsteve**

### CODE REVIEW

**Comment by Cadair** [comment]
@ehsteve I merged @wafels's PR, so can you resolve the conflicts now please?

**Comment by Cadair** [comment]
@ehsteve sorry I broke your PR agian :(

**Comment by Cadair** [comment]
@ehsteve this needs rebasing.

File: sunpy/net/dataretriever/downloader_factory.py [hunk 1]
```diff
@@ -28,12 +29,33 @@ def __init__(self, lst):
...
+    def __repr__(self):
+        return self.__str__()
+=======
     def _repr_html_(self):
         ret = ''
         for block in self:
             ret += block._repr_html_()
 
         return ret
+>>>>>>> upstream/unidown:sunpy/net/dataretriever/downloader_factory.py
```
Comment (dpshelio) [inline sunpy/net/dataretriever/downloader_factory.py]: conflicts errors to be cleaned up

File: sunpy/net/dataretriever/sources/noaa.py [hunk 2]
```diff
@@ -9,11 +9,17 @@
...
 
 class NOAAIndicesClient(GenericClient):
-    def _get_url_for_timerange(self, timerange, **kwargs):
+
+    @property
+    def _get_default_uri(self):
+        """Return the url to download indices"""
+        return ["ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt"]
+
+    def _get_url_for_timerange(cls, timerange, **kwargs):
```
Comment (Cadair) [inline sunpy/net/dataretriever/sources/noaa.py]: this should not be formatted like a class method any more.

File: sunpy/net/dataretriever/sources/noaa.py [hunk 3]
```diff
@@ -9,11 +9,17 @@
...
         """
         Helper function:
         """
-        return ["ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt"]
+        return NOAAIndicesClient._get_default_uri()
```
Comment (Cadair) [inline sunpy/net/dataretriever/sources/noaa.py]: should be `self`

File: sunpy/net/dataretriever/sources/noaa.py [hunk 4]
```diff
@@ -48,11 +54,16 @@ def _can_handle_query(cls, *query):
 
 class NOAAPredictClient(GenericClient):
 
-    def _get_url_for_timerange(self, timerange, **kwargs):
+    @classmethod
```
Comment (Cadair) [inline sunpy/net/dataretriever/sources/noaa.py]: same here.

File: sunpy/net/dataretriever/sources/noaa.py [hunk 5]
```diff
@@ -48,11 +54,16 @@ def _can_handle_query(cls, *query):
...
+    def _get_default_uri(cls):
+        """Return the url to download indices"""
+        return ["http://services.swpc.noaa.gov/text/predicted-sunspot-radio-flux.txt"]
+
+    def _get_url_for_timerange(cls, timerange, **kwargs):
         """
         Helper function:
         """
-        return ["http://services.swpc.noaa.gov/text/predicted-sunspot-radio-flux.txt"]
+        return NOAAPredictClient._get_default_uri()
```
Comment (Cadair) [inline sunpy/net/dataretriever/sources/noaa.py]: and here.

**Pull request merged**